src/lib/table/table-data-source.ts

```diff
@@ -97,22 +99,51 @@ export class MatTableDataSource<T> implements DataSource<T> {
    */
   sortingDataAccessor: ((data: T, sortHeaderId: string) => string|number) =
       (data: T, sortHeaderId: string): string|number => {
-    const value: any = data[sortHeaderId];
+    let value: any = data[sortHeaderId];
+    return isNumberValue(value) ? Number(value) : value;
+  }
 
-    // If the value is a string and only whitespace, return the value.
-    // Otherwise +value will convert it to 0.
-    if (typeof value === 'string' && !value.trim()) {
-      return value;
-    }
+  /**
+   * Returns a sorted array of the provided data according to the state of the MatSort. Called
```

Comment:
"Returns a sorted array..." -> "Gets a sorted copy of the data array based on the state of the MatSort"